### PR TITLE
Tests for `initialize_game` function

### LINKED ISSUE #86: [Feat][Surgeon Needed] Test `initialize_game`.
## Overview
Write all test cases and edge cases too when initializing a game. Please all cases that there is, and check if all associated parameters are properly initialized, and check for security risks too.

- This task assumes you're an expert in dojo tests
- Please you can ask questions, but please don't paste errors in my dm, thank you 😊.. Be like @pope-h 
- It assumes you have game experience and perhaps know about the poker game.
- Furthermore, it assumes you're a dev equal to or greater than me. 😋

## Furthermore?
- Your code must be statically typed for better readability please 🙏🏻
- Implement this test in `models/hand.cairo`
- Please ensure a hand works perfectly. Seems you just rank it yourself before asserting if the function equals your rank. Test for kicker's too. 🌚
- Document your code please, in a professional manner. Please avoid using too many small comments as you should know, only comment on hard to understand logic, and maybe some explanations on how and why it works
- When done, @ your username in a comment on top of the function you touched. If there's one there already, append with a comma and a space. (`// @ExistingUsername, @YourUsername`, etc)... It's cooler to leave your signature behind, yes? 😋
- Please avoid code repitition. For tests, there may be lines of codes that would be reused, please extract them into a separate function. Lock in your good PR quality please 🙏🏻
- May the force be with you ☪️✋🏻
- Ofcourse, if you have any questions (Which I bet you'd have, one billion per cent), you can branch [here](https://t.me/lemonade46) or [here](https://t.me/+tqBpITsr5mllZDQ0). Happy Hacks 😊💙

## Help?
If you want it, you can have it. Find it! I left everything this world has to offer _[there](https://t.me/+tqBpITsr5mllZDQ0)_ -- _**Pirate Gol D. Roger**_

### PR DESCRIPTION
**Pull request by SoarinSkySagar**
## Description   
Wrote tests for `initialize_game()`

## Related Issues   
closes #86   

## How to Test  
```bash
sozo test
``` 
  
## Checklist  
- [x] My code follows the project's coding style.  
- [x] I have tested these changes locally.

### CODE REVIEW

File: poker-texas-hold-em/contract/src/traits/player.cairo [hunk 1]
```diff
@@ -53,6 +53,22 @@ pub impl PlayerImpl of PlayerTrait {
...
+
+        if (game.id, game.reshuffled) != self.out {
+            game.players.append(self.id);
+        }
+        self.locked = (true, game.id);
+        self.in_round = true;
+        game.current_player_count += 1;
+        self.eligible_pots = 1;
+
+        game.current_player_count == game.params.max_no_of_players
```
Comment (Birdmannn) [inline poker-texas-hold-em/contract/src/traits/player.cairo]: Please this whole code is repeated from the `enter()` function. I have a few ideas...
1. update the `enter()` function, and add an extra parameter probably called `is_first_player` of type `bool`.
then from the `initialize` entry point, you call `player.enter(ref game, true)`, so that if `true`, you can skip some checks there, just as you skipped them in this new function.
2. Probably add a new field or use an existing field in the `game` to check if the player is the first player in the new game... then if true, skip the checks as said above.

Number two would be cleaner, but will take an overhead in code and tests adjustments (the overhead will only arise if you indeed added a new field to the `game`)
Number one is good, with a very tiny adjustments to just the two entry points that calls it. It's up to you ser.

**Review by Birdmannn — COMMENTED** [review]
Fairs. 🧘🏻‍♂️
Anyway, please check the review @SoarinSkySagar 

And yes, the `should_panic` tests, please remove the `ENTRYPOINT_FAILED` value from the `expected` panic statements.

That's all 
Thanks.
Sorry for the delay, by the way 🙇🏻

**Comment by SoarinSkySagar** [comment]
> Fairs. 🧘🏻‍♂️ Anyway, please check the review @SoarinSkySagar
> 
> And yes, the `should_panic` tests, please remove the `ENTRYPOINT_FAILED` value from the `expected` panic statements.
> 
> That's all Thanks. Sorry for the delay, by the way 🙇🏻

@Birdmannn actually the tests will fail if I remove them, as all the errors in the code give out the `ENTRYPOINT FAILED' error in addition to the original error. I have added them there for that, else the tests fails

**Comment by SoarinSkySagar** [comment]
@Birdmannn PTAL

**Comment by Birdmannn** [comment]
> > Fairs. 🧘🏻‍♂️ Anyway, please check the review @SoarinSkySagar
> > And yes, the `should_panic` tests, please remove the `ENTRYPOINT_FAILED` value from the `expected` panic statements.
> > That's all Thanks. Sorry for the delay, by the way 🙇🏻
> 
> @Birdmannn actually the tests will fail if I remove them, as all the errors in the code give out the `ENTRYPOINT FAILED' error in addition to the original error. I have added them there for that, else the tests fails

Ahh, I see

**Review by Birdmannn — APPROVED** [review]
Fairs 💯. Too easy.
Trés bien, Mister Sagar 🔥🔥
